ui-lib/src/main/java/co/electriccoin/zcash/ui/common/provider/AddressBookProvider.kt

```diff
@@ -23,19 +34,74 @@ class AddressBookProviderImpl : AddressBookProvider {
         file: File,
         addressBook: AddressBook
     ) {
-        file.outputStream().use {
-            serializeAddressBookToByteArrayFile(it, addressBook)
-        }
+        // TODO: Fetch AddressBookKey from keystore.
+        val addressBookKey = 0
+        encryptAddressBook(addressBookKey, file.outputStream(), addressBook)
     }
 
     override fun readAddressBookFromFile(file: File): AddressBook {
-        return file.inputStream().use {
-            deserializeByteArrayFileToAddressBook(it)
+        // TODO: Fetch AddressBookKey from keystore.
+        val addressBookKey = 0
+        return decryptAddressBook(addressBookKey, file.inputStream())
+    }
+
+    /**
+     * Serializes the given address book and then encrypts it for the given account.
+     *
+     * The ciphertext is written to the provided output stream.
+     */
+    private fun encryptAddressBook(
+        addressBookKey: AddressBookKey,
+        outputStream: OutputStream,
+        addressBook: AddressBook
+    ) {
+        // Generate a fresh one-time key for this ciphertext.
+        val salt = Random.randBytes(32)
+        val key = addressBookKey.deriveEncryptionKey(salt)
+
+        val plaintext =
+            ByteArrayOutputStream().use { stream ->
+                serializeAddressBook(stream, addressBook)
+                stream.toByteArray()
+            }
+
+        // Tink encodes the ciphertext as `nonce || ciphertext || tag`.
+        val cipher = ChaCha20Poly1305.create(key)
+        val ciphertext = cipher.encrypt(plaintext, null)
```

Comment:
```suggestion
        val ciphertext = cipher.encrypt(plaintext, null)
        if (ciphertext.size() != 12 + plaintext.size() + 16) {
            throw AssertionError(
                "unexpected ChaCha20Poly1305 ciphertext length ${ciphertext.size()} bytes" +
                    " for plaintext length ${plaintext.size()} bytes")
        }
```
(I do not use Kotlin's [assert](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/assert.html) function here because, like Java's `assert` keyword, it is disabled unless the `--enable-assertion`/`-ea` flag is passed.)